pupil_src/shared_modules/task_lib/background/shared_memory.py

```diff
@@ -0,0 +1,48 @@
+"""
+(*)~---------------------------------------------------------------------------
+Pupil - eye tracking platform
+Copyright (C) 2012-2018 Pupil Labs
+
+Distributed under the terms of the GNU
+Lesser General Public License (LGPL v3.0).
+See COPYING and COPYING.LESSER for license details.
+---------------------------------------------------------------------------~(*)
+"""
+
+
+class SharedMemory:
+    """
+    Contains information shared between the foreground and a background process.
+    All properties are process safe.
+    """
+
+    def __init__(self, should_terminate_flag, progress):
```

Comment:
On init `should_terminate_flag` and `progress` are shared cvalue objects while the equally-named properties are accessors for the `.value` attribute. This seems to be inconsistent. Maybe, SharedMemory should create the shared cvalue objects on its own and assign the arguments as values.

Reply:
Yes, good point!